package/MDAnalysis/coordinates/TPR.py

```diff
@@ -0,0 +1,80 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+
+from . import base
+from ..lib import util
+from .timestep import Timestep
+import MDAnalysis.topology.tpr.utils as tpr_utils
+import MDAnalysis.topology.tpr.setting as S
+
+
+import numpy as np
+
+
+class TPRReader(base.SingleFrameReaderBase):
+    # TODO: reduce duplication with `TPRparser`;
+    # we could share some state for the position
+    # in the binary file to avoid re-reading topology
+    # or perhaps combine the topology and coordinate reading
+    # with some inheritance shenanigans?
```

Comment:
I also feel that it would be good to avoid reading twice.

However, there's something to be said for starting with a clean separation, establishing correct functionality, and then thinking about optimization.

Reply:
Optional.